File: win-prog/ProgWinCS/Lines Curves and Area Fills/AntiAlias/AntiAlias.cs
//----------------------------------------
// AntiAlias.cs © 2001 by Charles Petzold
//----------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class AntiAlias: Form
{
     public static void Main()
     {
          Application.Run(new AntiAlias());
     }
     public AntiAlias()
     {
          Text = "Anti-Alias Demo";
          BackColor = SystemColors.Window;
          ForeColor = SystemColors.WindowText;
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Graphics grfx = pea.Graphics;
          Pen      pen  = new Pen(ForeColor);

          grfx.SmoothingMode   = SmoothingMode.None;
          grfx.PixelOffsetMode = PixelOffsetMode.Default;

          grfx.DrawLine(pen, 2, 2, 18, 10);
     }
}

File: win-prog/ProgWinVB/Lines Curves and Area Fills/AntiAlias/AntiAlias.vb
'------------------------------------------
' AntiAlias.vb (c) 2002 by Charles Petzold
'------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class AntiAlias
    Inherits Form

    Shared Sub Main()
        Application.Run(New AntiAlias())
    End Sub

    Sub New()
        Text = "Anti-Alias Demo"
        BackColor = SystemColors.Window
        ForeColor = SystemColors.WindowText
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        Dim pn As New Pen(ForeColor)

        grfx.SmoothingMode = SmoothingMode.None
        grfx.PixelOffsetMode = PixelOffsetMode.Default

        grfx.DrawLine(pn, 2, 2, 18, 10)
    End Sub
End Class
